flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateTransitions.java

```diff
@@ -72,6 +75,21 @@ void goToExecuting(
                 ExecutionGraph executionGraph,
                 ExecutionGraphHandler executionGraphHandler,
                 OperatorCoordinatorHandler operatorCoordinatorHandler);
+
+        /**
+         * Transitions into the {@link Executing} state.
+         *
+         * @param executionGraph executionGraph to pass to the {@link Executing} state
+         * @param executionGraphHandler executionGraphHandler to pass to the {@link Executing} state
+         * @param operatorCoordinatorHandler operatorCoordinatorHandler to pass to the {@link
+         *     Executing} state
+         * @param failureCollection collection of failures that are propagated
+         */
+        void goToExecuting(
```

Comment:
Can we do this without introducing a new method?

Reply:
We need this method for State classes that do not belong to StateWithExecutionGraph. Such as the WaitForResources -> Executing transition.

Alternatives:
1. Move the failureCollection Object into the AdaptiveScheduler instead of propagating and add methods to access it. I didn't do this, because I didn't want to make the AdaptiveScheduler bigger than it already is and this method would then also be accessible from outside.
2. Go to each state that transitions to Executing and add an empty List argument. 

I prefer the current solution to the alternatives.

Reply:
I'd prefer the option 2) as it might be tricky to keep the methods aligned otherwise and may lead to misuse.

It will also allow us to easily wrap "state that needs to be transferred" in the future.